### PR TITLE
PFIN-68 - Enhancements for ISS Interface v1.0.3 - new EMV tags

### PR DESCRIPTION
**Pull request by fsemerdzhieva**
added some additional checks+emv response data, merchantId and check for track2 length

### CODE REVIEW

File: ISS/develop/scripts/Build Request Message.js [hunk 1]
```diff
@@ -564,6 +577,7 @@ function getCurrencyConversionRate(fromValue, toValue) {
 
 function printObject(obj) {
     printObj(obj, 0);
+    //log.debug(getPrettyJsonString(obj));
```
Comment (dianavicol) [inline ISS/develop/scripts/Build Request Message.js:580]: Could you delete this?

File: ISS/develop/scripts/Build Request Message.js [hunk 2]
```diff
@@ -585,4 +599,13 @@ function printObj(obj, dep) {
...
 
+}
+
+//Format the JSON string
+function getPrettyJsonString(json) {
+    try {
+        return "\n" + JSON.stringify(json, null, 2);
+    } catch (e) {
+        return str;
+    }
```
Comment (dianavicol) [inline ISS/develop/scripts/Build Request Message.js:610]: And this

File: ISS/develop/scripts/Process Response Message.js [hunk 3]
```diff
@@ -114,10 +114,16 @@ function processCommonResponse(response) {
         txn.setTxnamountauthorised(txn.getTxnAmount());
         txn.setBillingamountauthorised(txn.getBillingamount());
     }
-
     // response.EMVResponse - optional
-    if (response.EMVResponse) {
-        // TODO KLAD
+    var emvData = response.commonResponse.EMVResponse;
+    if (response.commonResponse.EMVResponse) {
+        log.write("EMV");
```
Comment (dianavicol) [inline ISS/develop/scripts/Process Response Message.js:120]: Could you provide more details is this log, like _Processing EMV Response_

**Review by dianavicol — CHANGES_REQUESTED** [review]
Looks good. Just some teeny-tiny points

File: ISS/develop/scripts/Process Response Message.js (hunk 3 above)
Comment (dianavicol) [reply ISS/develop/scripts/Process Response Message.js:120]: and maybe remove the ToDo comment

Comment (fsemerdzhieva) [reply ISS/develop/scripts/Process Response Message.js:120]: Yeah this was debug comment I forgot to delete

**Review by dianavicol — APPROVED** [review]
👍

**Pull request merged**